datafusion/common/src/utils/mod.rs

```diff
@@ -291,10 +291,7 @@ pub(crate) fn parse_identifiers(s: &str) -> Result<Vec<Ident>> {
 }
 
 /// Construct a new [`Vec`] of [`ArrayRef`] from the rows of the `arrays` at the `indices`.
-pub fn get_arrayref_at_indices(
-    arrays: &[ArrayRef],
-    indices: &PrimitiveArray<UInt32Type>,
-) -> Result<Vec<ArrayRef>> {
+pub fn take_arrays(arrays: &[ArrayRef], indices: &dyn Array) -> Result<Vec<ArrayRef>> {
```

Comment:
just wondering how it differs from `arrow::compute::take`, looks like the same but for 2dim arrays instead of a single array.

What I'm thinking should we move this method to arrow-rs kernel?

Reply:
Exactly, this util is just to abstract away same outer iteration for the `compute::take`. If community thinks, this is beneficial, we can move this to `arrow-rs`. I think, this pattern is common enough to move it to `arrow-rs`.

Reply:
I think it makes sense to move to arrow. And maybe add a not int he docs for `RecordBatch` pointing people at it

It is common enough that it would be nice not to have to write it in each downstream crate